### PR TITLE
Re fetch membership > group field items after saving

### PR DESCRIPTION
**Pull request by damiankloip**
I think we can fix the @todo I left on the `OgMembershipItemList::get` method by just making sure we set `$fetched` to `FALSE` at the end of `::postSave`. If the items from the field are requested again after saving, during the same request.

Or are we fine with just leaving the field items that were populated from the widget? This way seems safer. But might be too safe.. Not sure. I guess it's possible that another user/request changed the groups too, so getting the current data again could be a good idea. Memberships are more likely to get lost otherwise?

### CODE REVIEW

**Comment by amitaibu** [comment]
I'm not sure I totally follow this part - maybe it's a good time to add tests? :)

**Comment by amitaibu** [comment]
>  maybe it's a good time to add tests?

Actually lets leave the tests for a separate issue.

**Comment by damiankloip** [comment]
Agree, tests in another issue would be good.

So I am thinking about this case (hopefully it makes sense) :
- User 1 adds group A and group B to the field
- User 2 adds just group C
- They save at pretty much the same time
- If the field values are used again in the request, both would not have A,B, and C. Just A and B, OR C

Now I talk about it, I think the chances are quite small. But this helps I think. This is no way as important and re-fetching the memberships in postSave, which we already do.

In a nutshell, this is just a simple change to try and make sure membership > group items in the fields are as current as possible after save.

**Comment by amitaibu** [comment]
You need to throw another factor there, which is access

So:
- User 1 adds group A and group B to the field
- User 2 adds just group C, but doesn't have access to group B

So the idea with the field is that it is just a way to pipe the accessible data to the user. It's important to thing of the group audience field as a _per-user_ - _per-request_. User 1 will see all groups; on the exact same time User 2 will see only A and C.

**Comment by damiankloip** [comment]
Yeah, we have not add any access checking yet. I think this makes this change make more sense? Refreshing could also get most recent access changes?

**Comment by amitaibu** [comment]
Needs a re-roll, and then you can merge. Thanks
